### ISSUE TITLE
ViTPose fails to train on a small dataset

### ISSUE THREAD

**Issue by chrisrapson** [ViTAE-Transformer/ViTPose#109]
Has anybody else had problems training on a small dataset? I've got one with 60 images in the training set, 10 in the validation set. It gives consistently 0 or -1 for all of the validation results.

I've tried:

- starting from a pretrained network, or from `pretrained=None`
- lowering the learning rate
- using a subset of the COCO dataset with 60/10 images, instead of my custom dataset

In all epochs, the results look like:
```
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] =  0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets= 20 ] =  0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets= 20 ] =  0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] =  0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] =  0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] =  0.000
 Average Recall     (AR) @[ IoU=0.50      | area=   all | maxDets= 20 ] =  0.000
 Average Recall     (AR) @[ IoU=0.75      | area=   all | maxDets= 20 ] =  0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] =  0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] =  0.000
2023-07-12 14:05:59,489 - mmpose - INFO - Epoch(val) [10][19]	AP: 0.0000, AP .5: 0.0000, AP .75: 0.0000, AP (M): 0.0000, AP (L): 0.0000, AR: 0.0000, AR .5: 0.0000, AR .75: 0.0000, AR (M): 0.0000, AR (L): 0.0000
```

or

```
2023-07-10 19:52:27,032 - mmpose - INFO - Epoch(val) [10][3]	AP: 0.0000, AP .5: 0.0000, AP .75: 0.0000, AP (M): -1.0000, AP (L): 0.0000, AR: 0.0000, AR .5: 0.0000, AR .75: 0.0000, AR (M): -1.0000, AR (L): 0.0000
```

It looks like ViTPose would usually report training results after 50 batches, but with a dataset this small, there are less than 50 batches per epoch.

Training on the full COCO dataset seems to be running fine. Starting from scratch, the `acc_pose` slowly increases from ~0.02 to ~0.3 over the first two epochs. Loss shrinks from 0.0022 to 0.0017.

**Comment by DuinoDu** [comment]
Your own custom data is also human pose data and num_keypoints is 17, as same order as coco keypoints?

**Comment by chris-rapson-formus** [comment]
My custom dataset was different, but as I said, I saw this strange behaviour even when training on a subset of the COCO dataset.

Anyway, it's a long time ago now. I managed to train a ViTPose network which had reasonable accuracy, but it turned out that other architectures were better for our use case.

**Comment by florent-enchantedtools** [comment]
Hi there. 
I have the same behavior here. On a custom dataset labeled with roboflow. 
It contains 290 images. The pose models contains 14 keypoints. Model input is 128x128 pxls.

```
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] =  0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets= 20 ] =  0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets= 20 ] =  0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] =  0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] =  0.000
 Average Recall     (AR) @[ IoU=0.50      | area=   all | maxDets= 20 ] =  0.000
 Average Recall     (AR) @[ IoU=0.75      | area=   all | maxDets= 20 ] =  0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] =  0.000
2025-02-23 21:57:58,243 - mmpose - INFO - Epoch(val) [200][2]	AP: 0.0000, AP .5: 0.0000, AP .75: 0.0000, AP (M): -1.0000, AP (L): 0.0000, AR: 0.0000, AR .5: 0.0000, AR .75: 0.0000, AR (M): -1.0000, AR (L): 0.0000
```

It seems that the loss(es) are decreasing.

```
2025-02-23 21:59:20,753 - mmpose - INFO - Epoch [230][7/7]	lr: 5.000e-06, eta: 0:11:59, time: 0.036, data_time: 0.005, memory: 2485, heatmap_loss: 0.0008, acc_pose: 1.0000, loss: 0.0008
```

I also note that tensorboard got no data even so a `events.out.tfevents.xxxxx` has been created. Dont know if this is connected.

Any idea ?
